### ISSUE TITLE
Spurious "Unexpected item made it into the items" warnings

### ISSUE THREAD

**Issue by ravenexp** [drahnr/cargo-spellcheck#271]
**Describe the bug**

Running `cargo-spellcheck -v` produces a lot of spurious warnings when run inside a project directory which is a git repo.

**To Reproduce**

Steps to reproduce the behaviour:

1. `cd` into a project directory which is a git repo.
2. Run `cargo spellcheck -v`
3. Get lots of warnings like:

```
[2022-08-17T10:55:38Z WARN  cargo_spellcheck::traverse] Unexpected item made it into the items /home/user/project/.git/objects/59/576875cf478ce9bcda9b9c996096da3c035123
```

**Expected behavior**

Files from `.git/` and directories listed in `.gitignore` are silently ignored.

**Please complete the following information:**
 - System: Arch Linux
 - Obtained: pacman
 - Version: cargo-spellcheck 0.11.2

**Comment by drahnr** [comment]
I cannot reproduce with latest `master`, could you provide a minimal test case?

**Comment by drahnr** [comment]
Could you retry the latest `master` - the warnings were a bit too focused on the use case around the use case in a cargo manifest dir - it's a _cargo_ plugin after all. I reduced the warning level and `.gitignore` is honoured now.
